### ISSUE TITLE
force_refresh_all_vehicles_states() - doesn't return odometer values

### ISSUE THREAD

**Issue by eili77** [Hyundai-Kia-Connect/hyundai_kia_connect_api#290]
* Hyundai / Kia Connect version: latest
* Python version: 3.11
* Operating System: Windows

### Description

Using `force_refresh_all_vehicles_states(self)` method doesn't return odometer values (all "none").
```
Vehicle(id='xxx', name='KONA', model='KONA', registration_date='2021-10-20 17:59:26.158', year=None, VIN='xxx', key=None, enabled=True, _total_driving_range=369.0, _total_driving_range_value=369.0, _total_driving_range_unit='km', _odometer=None, _odometer_value=None, _odometer_unit=None, _geocode_address=None, _geocode_name=None, car_battery_percentage=80, engine_is_running=False, last_updated_at=datetime.datetime(2023, 3, 6, 9, 2, 12, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Berlin')), timezone=None, dtc_count=None, dtc_descriptions=None, smart_key_battery_warning_is_on=False,...
```

Using `update_all_vehicles_with_cached_state(self)` works fine and returns odometer values.
```
Vehicle(id='xxx', name='KONA', model='KONA', registration_date='2021-10-20 17:59:26.158', year=None, VIN='xxx', key=None, enabled=True, _total_driving_range=394.0, _total_driving_range_value=394.0, _total_driving_range_unit='km', _odometer=22471.5, _odometer_value=22471.5, _odometer_unit='km', _geocode_address=None, _geocode_name=None, car_battery_percentage=85, engine_is_running=False, last_updated_at=datetime.datetime(2023, 3, 4, 16, 14, 21, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Berlin')), timezone=None, dtc_count=None, dtc_descriptions=None, smart_key_battery_warning_is_on=False,...
```

It's repeatable issue.
Is this a wanted/known behaviour?

**Comment by cdnninja** [comment]
Which region and brand?

**Comment by eili77** [comment]
Europe, Hyundai

**Comment by ZuinigeRijder** [comment]
I can confirm that force_refresh_all_vehicles_states() will NOT fill odometer. I digged into the differences into the response of:

````
DEBUG:hyundai_kia_connect_api.KiaUvoApiEU:hyundai_kia_connect_api - get_cached_vehicle_status response: {....
````

and

````
DEBUG:hyundai_kia_connect_api.KiaUvoApiEU:hyundai_kia_connect_api - Received forced vehicle data: {....
````

It appears that in Received forced vehicle data the following information is NOT returned:
- vehicleLocation
- odometer
- rest of the information appears to be in both responses

In the source code of KiaUvoApiEU.py the location is separately called, when force_refresh_all_vehicles_states() is called, to circumvent the absense of location (but nothing done for odometer).
````
DEBUG:hyundai_kia_connect_api.KiaUvoApiEU:hyundai_kia_connect_api - _get_location response: {'...
````

I do not know if this odometer was never returned in Received forced vehicle data. The workaround for the library user would be to call force_refresh_all_vehicles_states() first en then update_all_vehicles_with_cached_state(). Or KiaUvoApiEU.py should do this internally?

**Comment by ZuinigeRijder** [comment]
Because I tested with force_refresh_all_vehicles_states(), you can see well in the below Battery Monitor screenshot how much impact a force refresh has on your 12 volt battery.  At 8:07 I did the refresh. Your battery will be definitely drain if you use the refresh often. Especially when the SOC of your car is below 20%, because then the battery saver function will not kick in to top up your 12 volt battery, at least with the IONIQ 5. That is why my tool, [hyundai_kia_connect_monitor](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor) does no longer do a force refresh anymore.

![Screenshot_20230307-102514](https://user-images.githubusercontent.com/90704821/223380783-0c2b0acb-0aa8-45e3-80be-968dcb224e78.png)

**Comment by eili77** [comment]
> Because I tested with force_refresh_all_vehicles_states(), you can see well in the below Battery Monitor screenshot how much impact a force refresh has on your 12 volt battery. At 8:07 I did the refresh. Your battery will be definitely drain if you use the refresh often. Especially when the SOC of your car is below 20%, because then the battery saver function will not kick in to top up your 12 volt battery, at least with the IONIQ 5. That is why my tool, [hyundai_kia_connect_monitor](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor) does no longer do a force refresh anymore.
> 
> ![Screenshot_20230307-102514](https://user-images.githubusercontent.com/90704821/223380783-0c2b0acb-0aa8-45e3-80be-968dcb224e78.png)

Thanks for sharing these insights - understood :-)

Nevertheless, I think there's no obvious reason to omit the odometer information in this (force-) function, since all the other datapoints as far as I checked do exist...  - or am I wrong ?

**Comment by ZuinigeRijder** [comment]
@eili77 Yep, unfortunately the bluelink API does not return the odometer (and locations) values in the vehicles/[id]/status call, while it does in vehicles/[id]/status/latest
So then the library or the user has to work around it, by calling both after each other.

**Comment by eili77** [comment]
> Or do you see another solution?

Nope, unfortunately I don't. Thanks!

**Comment by cdnninja** [comment]
I thought I had this one fixed awhile back by not putting the value into vehicle unless provided.   This only works if you call cache regularly and occasionally call force.
